pkg/odo/cli/component/list.go

```diff
@@ -104,57 +144,120 @@ func (lo *ListOptions) Run() (err error) {
+		var err error
+
+		// TODO: wrap this into a component list for docker support
+		if lo.allAppsFlag {
+			dpl, err = lo.KClient.ListAllDeployments()
+
+		} else {
+			dpl, err = lo.KClient.ListDeployments(lo.Application)
+		}
+
 		if err != nil {
 			return err
 		}
 
-		var componentList []component.Component
+		if len(dpl.Items) != 0 {
+			lo.hasDevfileComponents = true
+			w := tabwriter.NewWriter(os.Stdout, 5, 2, 3, ' ', tabwriter.TabIndent)
+			fmt.Fprintln(w, "Devfile Components: ")
+			fmt.Fprintln(w, "APP", "\t", "NAME", "\t", "PROJECT", "\t", "TYPE", "\t", "REVISION")
+			for _, comp := range dpl.Items {
+				app := comp.Labels[applabels.ApplicationLabel]
+				cmpType := comp.Labels[componentlabels.ComponentTypeLabel]
+				revision := comp.Annotations["deployment.kubernetes.io/revision"]
+				fmt.Fprintln(w, app, "\t", comp.Name, "\t", comp.Namespace, "\t", cmpType, "\t", revision)
+			}
+			w.Flush()
+
+		}
+
+	}
+
+	// non-experimental workflow
 
-		if len(apps) == 0 && lo.LocalConfigInfo.ConfigFileExists() {
-			comps, err := component.List(lo.Client, lo.LocalConfigInfo.GetApplication(), lo.LocalConfigInfo)
+	// we now check if DC is supported
+	if lo.hasDCSupport {
+
+		var components component.ComponentList
+
+		if lo.allAppsFlag {
+			// retrieve list of application
+			apps, err := application.List(lo.Client)
 			if err != nil {
 				return err
 			}
-			componentList = append(componentList, comps.Items...)
-		}
 
-		// interating over list of application and get list of all components
-		for _, app := range apps {
-			comps, err := component.List(lo.Client, app, lo.LocalConfigInfo)
+			var componentList []component.Component
+
+			if len(apps) == 0 && lo.LocalConfigInfo.ConfigFileExists() {
+				comps, err := component.List(lo.Client, lo.LocalConfigInfo.GetApplication(), lo.LocalConfigInfo)
+				if err != nil {
+					return err
+				}
+				componentList = append(componentList, comps.Items...)
+			}
+
+			// interating over list of application and get list of all components
+			for _, app := range apps {
+				comps, err := component.List(lo.Client, app, lo.LocalConfigInfo)
+				if err != nil {
+					return err
+				}
+				componentList = append(componentList, comps.Items...)
+			}
+			// Get machine readable component list format
+			components = component.GetMachineReadableFormatForList(componentList)
+		} else {
+
+			components, err = component.List(lo.Client, lo.Application, lo.LocalConfigInfo)
 			if err != nil {
-				return err
+				return errors.Wrapf(err, "failed to fetch components list")
 			}
-			componentList = append(componentList, comps.Items...)
 		}
-		// Get machine readable component list format
-		components = component.GetMachineReadableFormatForList(componentList)
-	} else {
 
-		components, err = component.List(lo.Client, lo.Application, lo.LocalConfigInfo)
-		if err != nil {
-			return errors.Wrapf(err, "failed to fetch components list")
+		klog.V(4).Infof("the components are %+v", components)
```

Comment:
maybe remove this debug before merging

Reply:
This was added by tomas like 8 months ago https://github.com/openshift/odo/blob/master/pkg/odo/cli/component/list.go#L143, I can remove this if you want

Reply:
removed